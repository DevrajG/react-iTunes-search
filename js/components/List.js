import React from 'react';
import Item from './Item';

class List extends React.Component {

  render () {
    
    var res = this.props.res,
        results = res ? res.results.map((item) => {
          return (<Item key={item.trackId} data={item} />);
        }) : null;

    return (
      <div className="ui link cards">
        {results}
      </div>
    );
  }

}

export default List;