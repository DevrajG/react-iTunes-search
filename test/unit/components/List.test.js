import React from 'react';
import { shallow } from 'enzyme';
import List from '../../../src/components/List';

describe('<List />', () => {
  test('List component should renders correctly', () => {
    const props = {
      results: 2,
      resultCount: [
        {
          trackId: '1',
          trackPrice: 0.99,
          artworkUrl100: '',
          trackName: '',
          description: ''
        },
        {
          trackId: '2',
          trackPrice: 1.99,
          artworkUrl100: '',
          trackName: '',
          description: ''
        }
      ]
    };
    const component = shallow(<List {...props} />);
    expect(component).toMatchSnapshot();
  });
});
