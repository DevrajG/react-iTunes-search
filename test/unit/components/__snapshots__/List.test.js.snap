// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> List component should renders correctly 1`] = `
<div
  className="list-wrapper"
/>
`;
